### PR TITLE
Attempting to convert a jobbaned player offers control of their mob to ghosts

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Jobanning players from team antag will be a bit more effective now, without penalizing the person trying to convert them.

If no ghosts take the body the person lucks out and gets to dodge their jobban for a round.

### CODE REVIEW

**Comment by Firecage** [comment]
I do not support this. I would rather prefer that a random person doesn't get put in control of my body and that I instantly get ghosted.

**Comment by KorPhaeron** [comment]
@Firecage 

I would prefer if players followed the rules and didn't get jobbanned

**Comment by Jordie0608** [comment]
It's a good point however, just ejecting someone from the round is bad.

**Comment by KorPhaeron** [comment]
Letting the same player who griefed get the job to grief again is bad.

So is dealing with the people who refuse to play the role after getting converted and tell you to fuck off before returning to their job.

**Comment by Firecage** [comment]
Kor. If this gets merged, and I ever get rev banned. I would instantly suicide when Rev is announced, since I am not going to let someone else get control of my body, and I would be kicked from the round anyway.

**Comment by KorPhaeron** [comment]
@Firecage 

That would spare me the time and effort and risk of converting you only to have you grief or be unhelpful.

**Comment by Dorsisdwarf** [comment]
This is QOL for the people who have gone to the time and effort to convert shitters only to have them suicide /griff. :+1:

**Comment by KorPhaeron** [comment]
>time and effort to convert shitters only to have them suicide /griff. :+1:

Yeah this isn't as huge a deal in rev where you can just spam forever but kidnapping people in cult often takes significant effort/risk of getting caught, and in gang your conversion has a cooldown/delay.

**Comment by Bluespace** [comment]
Kor literally cannot stop coding good ideas. It must hurt to be so based.

**Comment by xxalpha** [comment]
Terrible idea, just prevent conversion.

**Comment by KorPhaeron** [comment]
@xxalpha 

So that they can scream "CULT" after you stun paper them and drag them onto a rune or go "REVS IN MEDBAY" after you flash them?

**Comment by xxalpha** [comment]
Make the conversion kill them then.

**Comment by KorPhaeron** [comment]
>stab someone with pen/flash them to convert them
>they fucking die

Great meta tell. And it'd still be pulling them straight out of the round and leaving you with a corpse for your conversion efforts. Everyone loses in that scenario.

**Comment by Firecage** [comment]
I would rather straight up die or be gibbed though instead of having a random person take control of my character.

**Comment by KorPhaeron** [comment]
@Firecage 

Won't be an issue if you don't break the rules. Lings have already been stealing your face for years, and the end round report will show it wasn't you.

**Comment by CosmicScientist** [comment]
If you're worried everyone's IC/meta opinion of your character/you will be abused, switch on random names/appearance or appeal to be unbanned from antag roles.

You shouldn't ruin everyone's fun by being a wasted convert. :+1:

**Comment by Bawhoppen** [comment]
This is stupid, It shouldn't put give away your character. If anything you ready up and its the round that has the antag you're banned from, it should just put you straight into observing as a ghost.

**Comment by xxalpha** [comment]
If you're worried about jobbanned people wasting conversions, make it so the antags can see who is not suitable for conversion.

**Comment by KorPhaeron** [comment]
@xxalpha 

So add an entire new HUD and a random number of immune crew members taking up who knows what job slots?

**Comment by bgobandit** [comment]
>offers control to ghost

THIS IS AWFUL WHY DOES IT HAVE TO TAKE THEM OUT OF THE ROOOUUUUNNND

>suggests that conversion just straight up kills them

**Comment by veganzombeh** [comment]
@bgobandit  Because leaving them in the round and not being converted actually *rewards* them for being jobbanned. This is the only fair way I can see to not punish the antags for their victims' jobban, while enforcing the jobban.

**Comment by lollerderby** [comment]
muh snowflakes

Just don't get job-banned.

**Comment by Bawhoppen** [comment]
Like I said, if say the gamemode is rev, and a player who is revbanned readies up, instead of spawning them at all, it just puts them into observe-mode

**Comment by bgobandit** [comment]
@veganzombeh I know I am mocking everyone else's objections to this PR that I support

**Comment by KorPhaeron** [comment]
@bawhoppen

I admit the ability to self ban and never spawn in shadowling or cult is tempting

**Comment by Firecage** [comment]
.....I must concur with you Kor. If I could self ban myself from cult and never spawn in a bloody cult round, I would be most happy. I hate team antag modes.

**Comment by Jalleo** [comment]
@bawhoppen that can be meta'd badly wont work.
Seriously this PR has the best solution that isnt too shitty or abusable.
Ghost becoming player converted antags is the worst form of antag converted characters, except for all of those that got banned

**Comment by Bawhoppen** [comment]
@bawhoppen how can it be metad at all even? and how won't it work either?

**Comment by Bawhoppen** [comment]
wait

I meant to reply to him, FUCK

**Comment by MrStonedOne** [comment]
We've been over this before, when ikky tried to make this.

if you make it so they don't join, then everybody who knows that person is job banned can use them being in the round as confirmation that it isn't that mode.

**Comment by Bawhoppen** [comment]
I see your point, but theres definitely some better solution than this PR.

**Comment by MrStonedOne** [comment]
No, there isn't.

We had a week long attempt to find this solution when ikky tried to make this exact same PR.

Your options are: fuck over the team antag, by either killing them or ghosting them without a replacement, fuck over the meta by making it easy to see what mode it is by who you know is job banned, or fuck over the banned player by giving his body to a ghost.

**Comment by Anonus** [comment]
You could have it give the ghost a random identity, though that might be a tell if someone is scouring the crewmember list and checking the annoucement system join messages to see if they match up.

**Comment by MrStonedOne** [comment]
There is only one other way.

Give everybody on the team antag a hud icon showing banned players, then deny the conversion.

**Comment by MrStonedOne** [comment]
old pr for reference: https://github.com/tgstation/-tg-station/pull/4440

@RemieRichards merged it, then @Aranclanos quick reverted it.

**Comment by optimumtact** [comment]
Good, maybe people will stop treating jobbans like a joke in future.

**Comment by Saegrimr** [comment]
Jobbans ARE a joke though.
Especially team antag bans, because they do literally nothing except from having you START as one.

If you don't want your special snowflake being sullied by a random ghost posessing you and squeezing your 3x3 pixel tits, don't get banned in the first place.

**Comment by Nanotransistor** [comment]
Could you make something similar for people who have the specific antag option disabled? Like say, i have gang turned off, and someone converts me, could this make a ghost take control of my body?

**Comment by KorPhaeron** [comment]
@Nanotransistor 

Maybe in another PR because this one is already pissing people off as is

**Comment by Supermichael777** [comment]
if the code is solid and it prevents those shits who instantly suicide when converted/deconverted from ruining everything then its fine by me.

File: code/game/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -533,3 +533,14 @@
 		return 0
 
 	return max(0, enemy_minimum_age - C.player_age)
+
+
+/datum/game_mode/proc/replace_jobbaned_player(mob/living/M, role_type)
+	var/list/mob/dead/observer/candidates = pollCandidates("Do you want to play as a [role_type]?", "[role_type]", null, FALSE, 100)
```
Comment (Shadowlight213) [inline code/game/gamemodes/game_mode.dm]: Use the special flag argument for pollCandidates so people who have that team antag turned off don't get the popup.

**Comment by SweaterKittens** [comment]
I like this. As much as I hate the idea of someone else taking your body - well it's even more of an incentive to not get banned.

**Comment by Steelpoint** [comment]
This is acceptable and fine, if you were job banned from a antagonist position then you lose all say in what happens to your body if your captured and converted.

Appeal your ban, don't fuck around to get banned, and try and avoid getting captured in the future. Or suicide and save everyone the trouble of capturing your banned ass and getting nothing out of it.

As oranges noted, job bans should have weight behind them and not be a joke.

**Comment by Robustin** [comment]
I approve. Anyone who thinks "make them immune to conversion" is a good idea should get their head checked.

**Comment by Bawhoppen** [comment]
I see this being problematic still, and the only 2 alternative solutions I could think of are
to force people with job/antagbans to have randomappearance and name, or
let them still be converted, but then just if they do anything shitty as the converted antag, give them a proper ban instead.

**Comment by Shadowlight213** [comment]
@Bawhoppen
How does forcing peoole with antag bans to have random appearance and name solve anything? We have appearance bans for that. Forcing someone to have a random appearance just for the possibility of it being a specific type of round is extremely shitty.
Your other alternative is basically current conversion antag bans, where the ban itself is a joke and little more than a note.

**Comment by KorPhaeron** [comment]
I don't get why people are so touchy about losing your character.

Why is this worse than changelings?

**Comment by Bawhoppen** [comment]
@Shadowlight213 It means their characters cannot be used if they have an antagban.

@KorPhaeron Personal characters are important, and a changeling impersonating you is not the same as a player playing as you.

**Comment by Shadowlight213** [comment]
@bawhoppen we have appearance bans that do that. Why would an antag ban have that functionality? It makes no sense.
If you don't want to lose your snowflake don't get banned.

**Comment by Bawhoppen** [comment]
@Shadowlight213 Its better than having your character be offered control by ghosts if you're converted into a team antag. But of the 2 suggestions I offered, the second is probably the most reasonable of all solutions people have mentioned which is the one we already use.

**Comment by Jordie0608** [comment]
It's not a cure-all solution, but is the most realistically viable and functional one that's come up.

**Comment by Iamgoofball** [comment]
You "muh character!!1" memers must throw a shitfit every time there's a changeling round

:+1: good change

**Comment by Bawhoppen** [comment]
@Iamgoofball they do throw a shitfit, but thats only because ling is shit, not because lings transform into anyone else

**Comment by KorPhaeron** [comment]
Has been updated to respect preferences

**Comment by MrStonedOne** [comment]
I'm MrStonedOne, and I approve this message.

**Comment by Jazaen** [comment]
So, now if you have conversion antag disabled your mob gets offered to a ghost on conversion?

**Comment by Steelpoint** [comment]
Ehh I'm a bit iffy on that one.

On the one hand it is good as it means if I hate a certain antag type that converts I can get out of it, but on the other hand what if I like the antag but I don't like being selected for it at round start?

Should be a separate PR, with more options.

**Comment by Jalleo** [comment]
I agree on preference thing being more of a config and wont that fuck over rounds if everyone sets it to no?

**Comment by Shadowlight213** [comment]
@Jazaen @Steelpoint @Jalleo
I don't think you understand what the pref check is doing. This will make it so that the popup won't show to ghosts if their pref is disabled. It doesn't affect living players that were converted.

**Comment by Jazaen** [comment]
Oh, sure, in this case it's great I suppose. Does it respect antag bans, so antagbanned person can't replace another antagbanned person?

**Comment by Jalleo** [comment]
@Shadowlight213 ah okay then
@Jazaen I think you cant get the popup when antagbanned since antag is turned off for you unless you get converted when nobody decides to become you

**Comment by Shadowlight213** [comment]
@KorPhaeron Merge conflicts

**Comment by KorPhaeron** [comment]
@Shadowlight213 

Well that sucks. I'll fix them if a maintainer tells me that this will be accepted eventually or not

File: code/game/gamemodes/cult/cult.dm [hunk 2]
```diff
@@ -187,6 +187,8 @@
 		cult_mind.current.cult_add_comm()
 		update_cult_icons_added(cult_mind)
 		cult_mind.current.attack_log += "\[[time_stamp()]\] <span class='danger'>Has been converted to the cult!</span>"
+		if(jobban_isbanned(cult_mind.current, "Cultist"))
```
Comment (RemieRichards) [inline code/game/gamemodes/cult/cult.dm]: use the role flags.
```ROLE_CULTIST```

Comment (KorPhaeron) [inline code/game/gamemodes/cult/cult.dm]: I thought the ROLE_CULTIST stuff was for prefs, not jobbans

Comment (RemieRichards) [inline code/game/gamemodes/cult/cult.dm]: tell that to the jobban calls that use it buddy.

Comment (KorPhaeron) [inline code/game/gamemodes/cult/cult.dm]: Oh, neat

**Comment by dumpdavidson** [comment]
Seems like a cool change. If people get anal about their original character do not steal being taken over you could always add a more visible note at the end of round log that the char was taken over.

**Comment by Bawhoppen** [comment]
Best solution still in the one we're currently doing, if you get converted to an antag you are banned from, and you fuck up again, you get a real ban

**Comment by duncathan** [comment]
Don't get antag banned

**Comment by Bawhoppen** [comment]
@duncathan >implying its just that simple

**Comment by duncathan** [comment]
@Bawhoppen I've never been banned; get on my level

**Comment by Bawhoppen** [comment]
@duncathan every ban ever placed has been just?

File: code/game/gamemodes/cult/cult.dm (hunk 2 above)
Comment (duncathan) [inline code/game/gamemodes/cult/cult.dm]: I missed this woops

**Pull request merged**